src/Backend/SgateShopgatePlugin/Components/Export.php

```diff
@@ -218,6 +237,72 @@ public function checkCompleteCategory($categoryId)
         return in_array($categoryId, $this->languageCompleteCategoryList);
     }
 
+    /**
+     * @return phpFastCache\Core\DriverAbstract
+     */
+    protected function getCacheInstance()
+    {
+        if (!$this->cacheInstance) {
+            $this->cacheInstance = CacheManager::getInstance(
+                'files', array(
+                    'path' => rtrim($this->config->getCacheFolderPath())
+                )
+            );
+        }
+
+        return $this->cacheInstance;
+    }
+
+    /**
+     * handles cache persistence for one catalog export
+     */
+    protected function initCache()
+    {
+        if (!isset($this->requestParams['offset']) || $this->requestParams['offset'] == 0) {
+            $this->getCacheInstance()->clean();
+        }
+    }
+
+    /**
+     * @param string $key
+     * @param string $value
+     */
+    protected function setExportCache($key, $value)
+    {
+        $instance   = $this->getCacheInstance();
+        $cachedData = $instance->get($key);
+
+        if ($cachedData && is_array($value)) {
+            $cachedData[key($value)] =  $value[key($value)];
```

Comment:
wait a second, why we using `key` function on a string?

Reply:
Good question to asked Alex, or we should find out ourselves.

Reply:
oh, thought I was reviewing your changes :)

Reply:
I mean, it will only be accessed if value is an array, but I don't know that scenario.